### PR TITLE
fix: status, members, online in chat header

### PR DESCRIPTION
**Pull request by iqmulator**

### CODE REVIEW

File: src/dotnet/Chat.UI.Blazor/Components/ChatHeader.razor [hunk 1]
```diff
@@ -26,7 +26,14 @@
...
             <div class="chat-header-title">
                 <div class="c-title">@chat.Title</div>
-                <StatusBadge ChatContext="@ChatContext" Colorful="@false"/>
+                <div class="c-info">
+                    @if (chat.Rules.CanJoin()) {
+                        @if (!chat.Id.IsPeerChat(out _)) {
+                            <ChatMemberCounter ChatContext="@ChatContext"/>
+                        }
+                    }
+                    <StatusBadge ChatContext="@ChatContext" Colorful="@false" ShowLastSeen="true"/>
```
Comment (frolyo) [inline src/dotnet/Chat.UI.Blazor/Components/ChatHeader.razor]: maybe IsColorful?

File: src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    protected override ComputedState<(int, int)>.Options GetStateOptions()
+        => ComputedStateComponent.GetStateOptions(GetType(),
+            static t => new ComputedState<(int, int)>.Options() {
+                InitialValue = (0, 0),
+                UpdateDelayer = FixedDelayer.NextTick,
+                Category = ComputedStateComponent.GetStateCategory(t),
+            });
+
+    protected override async Task<(int, int)> ComputeState(CancellationToken cancellationToken) {
```
Comment (frolyo) [inline src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor]: Let's have a Model as we always do for complex data

File: src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        var chatId = Chat.Id;
+        var allAuthorIds = await Authors.ListAuthorIds(Session, chatId, cancellationToken);
+
+        var authorPresences = await allAuthorIds
+            .Select(async authorId => await Authors.GetPresence(Session, chatId, authorId, cancellationToken).ConfigureAwait(false))
+            .Collect(256)
```
Comment (frolyo) [inline src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor]: why 256?

File: src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            .ConfigureAwait(false);
+
+        var onlineCount = authorPresences.Count(presence => presence is Presence.Away or Presence.Online or Presence.Recording);
+        return (allAuthorIds.Count(), onlineCount);
+    }
+
+    private string GetCount(int memberCount) {
```
Comment (frolyo) [inline src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor]: GetCountString or smth like this?

**Review by frolyo — CHANGES_REQUESTED** [review]

File: src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            default:
+                count = Math.Round(memberCount / 1000000m, 1, MidpointRounding.AwayFromZero);
+                return $"{count}M";
+        }
+    }
+
+    public sealed record Model {
+        public static readonly Model Loading = new();
+
+        public int MembersCount { get; init; }
```
Comment (frolyo) [inline src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor]: MemberCount or TotalCount

File: src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    <div class="chat-member-counter">
+        <div class="c-text">
+            <span>
+                @($"{GetCountString(m.MembersCount)} {"member".Pluralize(m.MembersCount)}")
+            </span>
+            @if (m.OnlineCount > 0) {
+                @($", {GetCountString(m.OnlineCount)} online")
+            }
+        </div>
+        <SeparatorDot/>
```
Comment (frolyo) [inline src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor]: What if we name it DotSeparator?

File: src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                UpdateDelayer = FixedDelayer.NextTick,
+                Category = ComputedStateComponent.GetStateCategory(t),
+            });
+
+    protected override async Task<Model> ComputeState(CancellationToken cancellationToken) {
+        var chatId = Chat.Id;
+        var allAuthorIds = await Authors.ListAuthorIds(Session, chatId, cancellationToken);
+
+        var authorPresences = await allAuthorIds
+            .Select(async authorId => await Authors.GetPresence(Session, chatId, authorId, cancellationToken).ConfigureAwait(false))
```
Comment (frolyo) [inline src/dotnet/Chat.UI.Blazor/Components/ChatMemberCounter/ChatMemberCounter.razor]: async await and ConfigureAwait can be omitted, right?

**Review by crui3er — APPROVED** [review]

**Review by frolyo — APPROVED** [review]

**Pull request merged**